account_invoice_import_invoice2data/models/invoice2data_template.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import base64
+import cgi
+import json
+import logging
+import os
+import pkg_resources
+import pprint
+import sys
+
+from openerp import _, api, fields, models
+from openerp.tools import file_open, float_compare
+from tempfile import mkstemp
+
+try:
+    from invoice2data.in_pdftotext import to_text
+    from invoice2data.main import extract_data
+    from invoice2data.main import logger as loggeri2data
+    from invoice2data.template import InvoiceTemplate
+    from invoice2data.template import read_templates
+    from invoice2data.utils import ordered_load
+except ImportError:
+    logging.error('Failed to import invoice2data')
+
+
+class Invoice2dataTemplate(models.Model):
+    _name = 'invoice2data.template'
+    _description = 'Template for invoice2data'
+
+    name = fields.Char(required=True)
+    template_type = fields.Selection([('purchase_invoice', 'Purchase Invoice')], 'Type',
+                                     required=False)
+    template = fields.Text(required=True)
+    preview = fields.Html()
+    preview_file = fields.Binary('File')
+    preview_text = fields.Html(readonly=True)
+    test_results = fields.Text(readonly=True)
+
+    @api.multi
+    def action_preview(self):
```

Comment:
Not a single documentation string? Try to get into the habit of documenting the code. Short meaningful sentences can bring such clarity to complex patterns.